Cargo.toml

```diff
@@ -44,6 +44,9 @@ miniz_oxide = { version = "0.7.0", default-features = false }
 addr2line = { version = "0.22.0", default-features = false }
 libc = { version = "0.2.146", default-features = false }
 
+[target.'cfg(not(any(all(windows, target_env = "msvc", not(target_vendor = "uwp")), target_os = "illumos")))'.dependencies]
```

Comment:
Why is this dependency added for uwp?

Reply:
It's not. Unless I screwed something up that part of the condition is copied from the section above. What's new here is that illumos is excluded as well.

Reply:
The conditional evaluates as follows on x86_64-uwp-windows-msvc:

* [x] `not(any(all(windows, target_env = "msvc", not(target_vendor = "uwp")), target_os = "illumos"))`
    * [ ] `any(all(windows, target_env = "msvc", not(target_vendor = "uwp")), target_os = "illumos")`
        * [ ] `all(windows, target_env = "msvc", not(target_vendor = "uwp"))`
            * [x] `windows`
            * [x] `target_env = "msvc"`
            * [ ] `not(target_vendor = "uwp")`
                * [x] `target_vendor = "uwp"`
        * [ ] `target_os = "illumos"`

So it will include zstd on x86_64-uwp-windows-msvc.

Reply:
Why is Windows included at all? Surely even gnu does not use elf sections due to the binary format being different on Windows?

Reply:
Sorry, you're right. This is what I get for commenting at 6:30 AM.

The uwp part was copied from the above section with miniz_oxide, addr2line, and libc. That exclusion was added in #543 without explanation.

The correct handling here would be for addr2line/libc to be gated on not using msvc (because mingw produces DWARF, not PDBs) and for miniz_oxide/zstd to be gated on not windows (because no Windows target uses ELF and these are for ELF compression).

Reply:
Ok I pushed a commit that cleans this up. Even if backtrace doesn't end up taking this PR it'll probably want something like that commit.